miso-dto/src/test/java/uk/ac/bbsrc/tgac/miso/dto/IlluminaNotificationDtoTest.java

```diff
@@ -0,0 +1,107 @@
+
+
+  @Test
+  public void testPartiallyPopulatedIlluminaNotificationRoundTrip() throws Exception {
+    IlluminaNotificationDto notificationDto = new IlluminaNotificationDto();
+    notificationDto.setSequencerName("Coffee");
+    notificationDto
+        .setCompletionDate(ZonedDateTime.of(LocalDateTime.of(LocalDate.of(2017, 2, 23), LocalTime.of(0, 0)), ZoneId.of("America/Toronto")));
+    notificationDto.setHealthType(HealthType.Started);
+
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new JavaTimeModule())
+        .setDateFormat(new ISO8601DateFormat())
+        .enable(SerializationFeature.INDENT_OUTPUT);
+    String serialized = mapper.writeValueAsString(notificationDto);
+
+    NotificationDto deSerialized = mapper.readValue(serialized, NotificationDto.class);
+    assertThat("Round trip of", notificationDto, is(deSerialized));
+  }
+
+  @Test
+  public void testFullyPopulatedIlluminaNotificationRoundTrip() throws Exception {
+    IlluminaNotificationDto notificationDto = fullyPopulatedIlluminaNotificationDto("RUN_B");
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new JavaTimeModule())
+        .setDateFormat(new ISO8601DateFormat())
+        .enable(SerializationFeature.INDENT_OUTPUT);
+    String serialized = mapper.writeValueAsString(notificationDto);
+
+    NotificationDto deSerialized = mapper.readValue(serialized, NotificationDto.class);
+    assertThat("Round trip of", notificationDto, is(deSerialized));
+  }
+
+  /**
+   * Demonstrations day off by one error when MISO and RunScanner are in different time zones separated by midnight.
+   * 
+   * The RunScanner is in Toronto.
+   * 
+   * MISO is in Winnipeg, one hour behind Toronto.
+   * 
+   * The sequencer run RUN_B finishes at midnight in Toronto.
+   * 
+   * Once the date is transferred to Winnipeg it's one hour behind, which puts it in the previous day. This is demonstrated in the test by
+   * the necessity to add one day to the Winnipeg date in order for the values to match.
+   * 
+   * @throws Exception
+   */
+  @Test
+  public void testMidnightTimezoneShenanigans() throws Exception {
+    IlluminaNotificationDto notificationDto = fullyPopulatedIlluminaNotificationDto("RUN_B");
+    ObjectMapper mapper = new ObjectMapper();
+    mapper.registerModule(new JavaTimeModule())
+        .setDateFormat(new ISO8601DateFormat())
+        .enable(SerializationFeature.INDENT_OUTPUT);
+    String serialized = mapper.writeValueAsString(notificationDto);
+    NotificationDto deSerialized = mapper.readValue(serialized, NotificationDto.class);
+    
+    ZonedDateTime runScannerTorontoStartDate = notificationDto.getStartDate().withZoneSameInstant(ZoneId.of("America/Toronto"));
+    ZonedDateTime misoWinnipegStartDate = deSerialized.getStartDate().withZoneSameInstant(ZoneId.of("America/Winnipeg"));
+
+    assertThat(runScannerTorontoStartDate.toLocalDate(), is(misoWinnipegStartDate.toLocalDate().plusDays(1)));
```

Comment:
This should be considered a failure. We don't want the date to change.

MISO only saves a date (no time), so the notification server/run scanner is only responsible for finding the date (no time). Currently, the `IlluminaTransformer` parses the startDate from the run directory. It will always assume midnight, so we'll end up with a run named '170607...' having a startDate of 170606 whether the run started at 00:00 or 23:59.

To test whether this is going to work in MISO requires converting to a Run, where the startDate will be a java.util.Date, and ensuring that it is the same date as specified by whatever class is used in the NotificationDto.